### PR TITLE
Remove jaxrs classes from client

### PR DESCRIPTION
**Pull request by carlesarnal**

### CODE REVIEW

**Comment by EricWittmann** [comment]
There were CI build failures.  I've triggered the build to re-run just in case.  I didn't dig into the reasons for the failures.  Also note that with another PR being merged you have a conflict in this now.  Sorry about that. :)

File: rest-client/src/main/java/io/apicurio/registry/client/RegistryRestClient.java [hunk 1]
```diff
@@ -34,7 +34,7 @@
     ArtifactMetaData createArtifact(ArtifactType xRegistryArtifactType, String xRegistryArtifactId, IfExistsType ifExists, InputStream data);
 
 
-    Response getLatestArtifact(String artifactId);
+    ResponseBody getLatestArtifact(String artifactId);
```
Comment (EricWittmann) [inline rest-client/src/main/java/io/apicurio/registry/client/RegistryRestClient.java:37]: I was thinking maybe `byte[]` or `InputStream` or `ContentHandle` here.  Is there something that the okhttp3 class gives us that's better than just plain data?

Comment (carlesarnal) [reply rest-client/src/main/java/io/apicurio/registry/client/RegistryRestClient.java:37]: You can get the `InputStream` from the `ResponseBody` class and also the content length and type.

**Comment by carlesarnal** [comment]
> There were CI build failures. I've triggered the build to re-run just in case. I didn't dig into the reasons for the failures. Also note that with another PR being merged you have a conflict in this now. Sorry about that. :)

Yes, the `ClusterIT` is failing for the infinispan storage variant, I'll take a look.

File: utils/maven-plugin/src/main/java/io/apicurio/registry/maven/DownloadRegistryMojo.java [hunk 2]
```diff
@@ -117,17 +116,13 @@ protected void executeInternal() throws MojoExecutionException {
...
-                    }
-                } finally {
-                    response.close();
+                InputStream stream = (version != null) ?
+                        getClient().getArtifactVersion(version, id) :
+                        getClient().getLatestArtifact(id);
+                if (replaceExisting) {
+                    Files.copy(stream, outputFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
+                } else {
+                    Files.copy(stream, outputFile.toPath());
```
Comment (EricWittmann) [inline utils/maven-plugin/src/main/java/io/apicurio/registry/maven/DownloadRegistryMojo.java:125]: Can you confirm that `Files.copy()` closes the stream?

**Review by EricWittmann — APPROVED** [review]

File: utils/maven-plugin/src/main/java/io/apicurio/registry/maven/DownloadRegistryMojo.java (hunk 2 above)
Comment (carlesarnal) [reply utils/maven-plugin/src/main/java/io/apicurio/registry/maven/DownloadRegistryMojo.java:125]: Good catch, no, it does not close the stream.

**Review by EricWittmann — APPROVED** [review]

**Comment by EricWittmann** [comment]
Nice.  Thanks @carlesarnal

**Pull request merged**